vispy/visuals/markers.py

```diff
@@ -197,351 +202,260 @@
+                spike_y = sinangle * x + cosangle * (y + shift_y);
+            }
+            // in the frame where the spike is upwards:
+            // rotate 18 deg the zone x < 0 around the top of the star
+            // (point whose coords are -s/2, 0 where s is the size of the marker)
+            // compute y coordonates as well because
+            // we do a second rotation to put the spike at its final position
+            float rot_center_y = -$v_size/2.;
+            float rot18x = cos(PI2_20) * spike_x
+                                - sin(PI2_20) * (spike_y - rot_center_y);
+            //rotate -18 deg the zone x > 0 arount the top of the star
+            float rot_18x = cos(PI2_20) * spike_x
+                                + sin(PI2_20) * (spike_y - rot_center_y);
+            float bottom = spike_y - $v_size/10.;
+            //                     max(left edge, right edge)
+            float spike = max(bottom, max(rot18x, -rot_18x));
+            if (i == 0)
+            {// first spike, skip the rotation
+                star = spike;
+            }
+            else // i > 0
+            {
+                star = min(star, spike);
+            }
         }
-        vec2 normal = vec2(2.*x0/v_size.x/v_size.x, 2.*y0/v_size.y/v_size.y);
-        norm = length(normal);
-    }
-    return (f - 0.5) / norm;
+        return star;
+    """,
+
+    'cross_lines': """
+        //vbar
+        float r1 = abs(pointcoord.x - 0.5)*size;
+        float r2 = abs(pointcoord.y - 0.5)*size - $v_size/2;
+        float vbar = max(r1,r2);
+        //hbar
+        float r3 = abs(pointcoord.y - 0.5)*size;
+        float r4 = abs(pointcoord.x - 0.5)*size - $v_size/2;
+        float hbar = max(r3,r4);
+        return min(vbar, hbar);
+    """,
 }
-"""
 
-cross_lines = """
-float cross(vec2 pointcoord, float size)
-{
-    //vbar
-    float r1 = abs(pointcoord.x - 0.5)*size;
-    float r2 = abs(pointcoord.y - 0.5)*size - $v_size/2;
-    float vbar = max(r1,r2);
-    //hbar
-    float r3 = abs(pointcoord.y - 0.5)*size;
-    float r4 = abs(pointcoord.x - 0.5)*size - $v_size/2;
-    float hbar = max(r3,r4);
-    return min(vbar, hbar);
+# combine all the symbol shaders in a big if-else statement
+symbol_func = f"""
+float symbol(vec2 pointcoord, float size, int symbol) {{
+   {' else'.join(
+    f''' if (symbol == {i}) {{
+        // {name}
+        {shader}
+    }}'''
+    for i, (name, shader) in enumerate(symbol_shaders.items())
+    )}
+}}"""
+
+# aliases
+symbol_aliases = {
+    'o': 'disc',
+    '+': 'cross',
+    '++': 'cross_lines',
+    's': 'square',
+    '-': 'hbar',
+    '|': 'vbar',
+    '->': 'tailed_arrow',
+    '>': 'arrow',
+    '^': 'triangle_up',
+    'v': 'triangle_down',
+    '*': 'star',
 }
-"""
+
+symbol_shader_values = {name: i for i, name in enumerate(symbol_shaders)}
+symbol_shader_values.update({
+    **{alias: symbol_shader_values[name] for alias, name in symbol_aliases.items()},
+})
```

Comment:
This is where the real changes are: an autogenerated if-else chain that dispatches to the right symbol. The big diff above this is only moving code from isnide individual functions to a dictionary of snippets, but the code itself is identical.